src/MatBlazor/Components/MatAutocompleteList/MatAutocompleteList.razor

```diff
@@ -9,7 +9,7 @@
     @if (IsShowingClearButton)
     {
         <div class="mat-autocomplete-list-clearbutton">
-            <MatIconButton Icon="clear" OnMouseDown="@ClearText"></MatIconButton>
+            <MatIconButton Icon="clear" OnMouseDown="@ClearText" OnClick="@ClearText"></MatIconButton>
```

Comment:
Using the OnClick event is working both for keyboard input and mouse keyboard and OnMouseDown is not needed anymore. If it's still handled the ClearText method will be called twice. Please, remove the OnMouseDown handler.